modules/nic/variables.tf

```diff
@@ -0,0 +1,47 @@
+variable "name" {
+  description = "The name of this network interface card."
+  type        = string
+}
+
+variable "resource_group_name" {
+  description = "The name of the resource group to create the resources in."
+  type        = string
+}
+
+variable "location" {
+  description = "The location to create the resources in."
+  type        = string
+}
+
+variable "ip_configuration" {
+  description = "A map of IP configuration with supported blockss"
+
+  type = map(object({
+    name                          = string
+    subnet_id                     = optional(string)
+    private_ip_address_version    = optional(string)
+    private_ip_address_allocation = string
+    private_ip_address            = optional(string)
+    primary                       = optional(bool, true)
```

Comment:
Suggest putting the required arguments first and also writing the comments behind the related arguments as a "description". This is to make it easier to read as documentation.

Instead of:

```terraform
# private ip address version defaults to IPv4, and subnet id is required when private ip address version is set to IPv4
# got the private ip address from a created network interface in Azure portal and following that example for the complete example
```

Try:

```suggestion
  type = map(object({
    name                          = string
    private_ip_address_allocation = string
    private_ip_address_version    = optional(string) # Possible values are `IPv4` or `IPv6`. Defaults to `IPv4`.
    subnet_id                     = optional(string) # Required when `private_ip_address_version` is set to `IPv4`.
    private_ip_address            = optional(string)
    primary                       = optional(bool, true)
```

Reply:
Also look into if other arguments should be commented as well.

Reply:
the requirement are now put first ahead of the optional arguments and they have been commented aswell